sku.0/sys.server/compiled/game/script/systems/combat/combat_actions.java

```diff
@@ -10483,21 +10478,24 @@ public int bm_revive_pet_1(obj_id self, obj_id target, String params, float defa
             setCommandTimerValue(self, TIMER_COOLDOWN, 0.0f);
             return SCRIPT_OVERRIDE;
         }
-        if (getDistance(getLocation(self), getLocation(beast)) > 7.0f)
+        if (!combatStandardAction("bm_revive_pet_1", self, target, params, "", ""))
         {
-            sendSystemMessage(self, new string_id("spam", "stand_next_to_corpse"));
             setCommandTimerValue(self, TIMER_COOLDOWN, 0.0f);
             return SCRIPT_OVERRIDE;
         }
-        if (!combatStandardAction("bm_revive_pet_1", self, target, params, "", ""))
-        {
+        float playerToPetRange = getDistance(getLocation(self), getLocation(beast));
+        if (playerToPetRange > 84f) {
+            sendSystemMessage(self, new string_id("spam", "stand_next_to_corpse"));
             setCommandTimerValue(self, TIMER_COOLDOWN, 0.0f);
             return SCRIPT_OVERRIDE;
-        }
-        messageTo(self, "channelRevivePet", null, 0, false);
+        } 
+        float percentRangeMod = (playerToPetRange > 7f) ? (playerToPetRange / 84f) * 10f : 0f ; 

```

Comment:
Should the false value in the ternary be 1f not 0f? If we multiply by 0f we are always setting the cooldown time to 0 instead of the difference between the baseCooldownTime and cooldownTimeMod.

Reply:
Also updated this to 85 and changed it to default to 1